vowpalwabbit/cb_algs.cc

```diff
@@ -60,7 +61,10 @@ void predict_or_learn(cb& data, single_learner& base, example& ec)
     ec.l.cs = data.cb_cs_ld;
 
     // Guard example state restore against throws
-    auto restore_guard = VW::scope_exit([&ld, &ec] { ec.l.cb = ld; });
+    auto restore_guard = VW::scope_exit([&ld, &ec] {
+      ec.l.cs.costs = v_init<COST_SENSITIVE::wclass>();
```

Comment:
why do we need to clear the costs in case of a throw?

Reply:
This isn't in case of a throw, this is something that must happen every time this function exits. If we don't reset it, it results in a double free.

Reply:
ah, right. keep getting misled by that comment and forget it's a scope guard at first

Reply:
should this not be `ec.l.cs.costs.clear()`?

Reply:
Clear simply removes all the elements but still refers to the buffer. We actually want to completely remove the reference to the buffer here.

Reply:
I am guessing the response is the same as: https://github.com/VowpalWabbit/vowpal_wabbit/pull/2275/files#r554070465

Reply:
Yep :)